src/libcore/iter/sources.rs

```diff
@@ -31,8 +32,41 @@ impl<A: Clone> Iterator for Repeat<A> {
 
     #[inline]
     fn next(&mut self) -> Option<A> { Some(self.element.clone()) }
+
     #[inline]
     fn size_hint(&self) -> (usize, Option<usize>) { (usize::MAX, None) }
+
+    #[inline]
+    fn nth(&mut self, _: usize) -> Option<A> { self.next() }
+
+    #[inline]
+    fn all<F>(&mut self, f: F) -> bool where F: FnMut(A) -> bool { self.any(f) }
```

Comment:
I'm confused by this one, since it looks like it can still be infinite.  If we want this kind of behavior, wouldn't it be `f(self.element.clone())`?